src/aviary/dataset_server.py

```diff
@@ -0,0 +1,211 @@
+import asyncio
+import logging
+import time
+import traceback
+import uuid
+from contextlib import contextmanager
+from itertools import starmap
+from typing import Generic, TypeVar
+
+import uvicorn
+from pydantic import BaseModel, Field
+
+from aviary.env import Environment, TaskDataset
+from aviary.tools import MessagesAdapter, ToolRequestMessage, ToolsAdapter
+
+try:
+    from fastapi import FastAPI, HTTPException
```

Comment:
Can you add `fastapi` to `pre-commit`'s `mypy`

Reply:
Already there: https://github.com/Future-House/aviary-internal/blob/9c82e81bbc2c77d30bcaf007d84f13c9a72c9cd8/.pre-commit-config.yaml#L64

Reply:
That is the wrong repo tho, this is aviary, not aviary-internal

Reply:
Should probably also add `uvicorn` too

Reply:
Oops 🤦🏾. Fixing